### PR TITLE
Improvements to interaction tools

### PR DESCRIPTION
**Pull request by ACrazyTown**
Here it is, should be everything like described EXCEPT toggling with shift. I bound it to left alt instead because I couldn't figure out how to get shift to work due to the mover's shortcut.

Lmk if any changes are neccessary

### CODE REVIEW

File: flixel/system/debug/interaction/tools/Pointer.hx [hunk 1]
```diff
@@ -21,13 +21,17 @@ class GraphicCursorCross extends BitmapData {}
  */
 class Pointer extends Tool
 {
+	public var _selectionArea:FlxRect = new FlxRect();
```
Comment (Geokureli) [inline flixel/system/debug/interaction/tools/Pointer.hx:24]: Why was this made public? your not using it in any external classes

File: flixel/system/debug/interaction/tools/Pointer.hx [hunk 2]
```diff
@@ -21,13 +21,17 @@ class GraphicCursorCross extends BitmapData {}
...
+
 	var _selectionStartPoint:FlxPoint = new FlxPoint();
 	var _selectionEndPoint:FlxPoint = new FlxPoint();
 	var _selectionHappening:Bool = false;
 	var _selectionCancelled:Bool = false;
-	var _selectionArea:FlxRect = new FlxRect();
+	var _selectedItem:FlxBasic;
```
Comment (Geokureli) [inline flixel/system/debug/interaction/tools/Pointer.hx:30]: this field is set but never used

**Review by Geokureli — CHANGES_REQUESTED** [review]
I'm not seeing the functionality that was discussed. I expected that pressing an object while holding alt would add or remove it from the selected objects list. whenever I click anywhere with alt pressed it seems to deselect every object.

File: flixel/system/debug/interaction/tools/Pointer.hx (hunk 1 above)
Comment (ACrazyTown) [reply flixel/system/debug/interaction/tools/Pointer.hx:24]: My bad, I forgot to remove it after testing something

File: flixel/system/debug/interaction/tools/Pointer.hx (hunk 2 above)
Comment (ACrazyTown) [reply flixel/system/debug/interaction/tools/Pointer.hx:30]: Same with this, forgot to remove old debugging stuff

**Comment by ACrazyTown** [comment]
I'll convert this to a draft until I figure out everything

**Comment by Geokureli** [comment]
> > I expected that alt+clicking an object would add or remove it from the selected objects list.
> 
> By the way, is this not what holding CTRL does? Or am I misunderstanding what's supposed to be done?

Ctrl+Click seems to do nothing for me. Maybe it's because I'm on a mac keyboard and perhaps I'm seeing different behavior? can you descript what behavior you're seeing?

**Comment by ACrazyTown** [comment]
> > > I expected that alt+clicking an object would add or remove it from the selected objects list.
> > 
> > By the way, is this not what holding CTRL does? Or am I misunderstanding what's supposed to be done?
> 
> Ctrl+Click seems to do nothing for me. Maybe it's because I'm on a mac keyboard and perhaps I'm seeing different behavior? can you descript what behavior you're seeing?

It's similar to what you described, while holding Ctrl and clicking items get added/removed to the list. I think its a Mac issue because I had a similar one too, not with Flixel but with Ctrl clicking

**Comment by Geokureli** [comment]
I'll grab my win pc later and try

**Comment by ACrazyTown** [comment]
@Geokureli Did you try? Don't want to be annoying, just want to know before I start making further changes

**Comment by Geokureli** [comment]
I will check today, sorry

**Comment by Geokureli** [comment]
@ACrazyTown Yes, it seems like Ctrl+click behaves exactly like I was imagining Shift would behave. I should probably try to make this work on non-windows machines

**Comment by ACrazyTown** [comment]
> @ACrazyTown Yes, it seems like Ctrl+click behaves exactly like I was imagining Shift would behave. I should probably try to make this work on non-windows machines

Alright, I will remove my attempt at making the shift click. Should I rebind Ctrl to Shift?

**Comment by Geokureli** [comment]
I think I have a lot to learn about the current features before I can say for sure what I want. lets close this for now unless theres things you really think would be beneficial. feel free to reopen, later

**Pull request closed without merging**